# Project Group 2 

Members: Jelmer Nederhof, Yvain Redon, Martin Kooistra, Chenhao Zhang and Suze Vernooij

Student numbers: 5407400, 6071392, 4880005, 5704057 and 5091292

# Research Objective

*Requires data modeling and quantitative research in Transport, Infrastructure & Logistics*

Reveal the relationship between covid and public transport patterns for different sociodemographic groups in the Netherlands between 2010 and 2022.

**- RQ should include data processing, quantitative analysis and visualisation.**
Reveal the relationship between covid and public transport patterns for different sociodemographic groups in the Netherlands between 2010 and 2022.

**- Provide the data source you need for answering the RQ, provide link to the data where
applicable**

* https://opendata.cbs.nl/#/CBS/en/dataset/84707ENG/table
* https://opendata.cbs.nl/statline/#/CBS/nl/dataset/83496NED/table?ts=1696408326201

**- Provide information about the time scale and spatial scale you are looking at. Provide the
geographical boundary of your study.**

The scope of the project is within the Netherlands. Data from 2010 until 2022 will be used.

# Contribution Statement

*Be specific. Some of the tasks can be coding (expect everyone to do this), background research, conceptualisation, visualisation, data analysis, data modelling*

04/10/2023

Jelmer Nederhof, Yvain Redon, Martin Kooistra and Suze Vernooij worked on the project proposal. Chenhao Zhang will join later since he was added to our group later.

**Author 1**:

**Author 2**:

**Author 3**:

**Author 4**:

**Author 5**:


# Data Used

- https://opendata.cbs.nl/#/CBS/en/dataset/84707ENG/table
- https://opendata.cbs.nl/statline/#/CBS/nl/dataset/83496NED/table?ts=1696408326201

# Data Pipeline

Define the data sources: Identify where your data comes from and how it will be collected.

Clean and validate data: Use Python libraries like Pandas and NumPy to clean, validate, and preprocess your data.

Transform and enrich data: Apply data transformations and enrichments to improve the quality of your data for analysis.

Store the processed data: Store the processed data in a suitable storage system, such as a database or cloud storage.

Analyze and visualize data: Use Python libraries such as Matplotlib, Seaborn, and Plotly for data visualization and analysis.




### Introduction

In 2020, the COVID-19 pandemic set foot in the Netherlands resulting in infections all over the country. To decrease the infections, the Dutch government took several measurer. These sets of policies changed the behavioral patterns of Dutch citizens. A new trend was to work from home resulting in less commuting and less mobility. Policies and trends like these effected the use public transport. In this report, the following research question will be discussed: 
“What is the relationship between the COVID-19 pandemic and the use of public transport different sociodemographic groups in the Netherlands between 2010 and 2022?”

### Methodology

To reveal this relationship, datasets about the use of public transport are analyzed. For each analyzing step, a sub question is formulated to answer the research question. 

Two datasets from CBS are used to analyze. One covers data of the usage of public transport. This percentage is estimated by asking people in the survey if they traveled by public transport that day. One dataset has data from 2010 to 2017 and the other one from 2018 to 2022. 
First step is to merge this data together to one dataset by setting the same language, changing variable names and sorting the years. 

In the survey the dataset is based on, questions were asked about social-demographic characteristics of the ones making the survey. Using Python, the dataset is analyzed to get to know patterns and to see if these socio-demographic characteristics influenced these patterns. 
Multiple sub questions have been formulated to specify the research question better. The dataset is used to find general patterns before the pandemic. Graphs will specify the trend over the years:

*1.	How did the COVID-19 pandemic impact overall public transport ridership in the Netherlands between 2020 and 2022?*

The trends for different groups can be plotted in one graph. This graph can be used to see were the biggest changes happened during the pandemic.

*2.	What were the key sociodemographic factors which were impacted most during the pandemic?*

Data visualization for these groups can specify the trend more. When needed, literature review can be a tool to explain why these changes happened over time. 

*3.	How did these key sociodemographic factors changed during the pandemic?*

[..….We did coding from this point onwards. We are thinking about adding other groups (e.g. level of education). We created more sub questions to expand the research further (e.g. forecasts)……] 

More recent data about the use of public transport after the pandemic, can be compared with the data before the pandemic and before. If the data is close to the pre-covid data, the public transport sector returned back to pre-pandemic patterns. Also, other analyzing methods can create forecasts on how the current trend will recover from the pandemic in the future. 

*4.	Did public transport usage patterns return to pre-pandemic levels for different sociodemographic groups in the Netherlands by 2022, and if not, what are the main determinants of these disparities?*

The dates when the government implemented new policies can be considered while analyzing the dataset. For example, during the first lockdown it’s expected that people commute less. 

*5.	How did government policies, such as lockdowns, mask mandates, and vaccination campaigns, influence public transport patterns for various sociodemographic groups during the pandemic years?*





In [5]:
import pandas as pd
data = pd.read_csv('Project_datasets_merged_filtered.csv')
data.head()

Personal characteristics  Topic  Traffic participation  \
0      Age: 12 to 17 years   2010                   82.1   
1      Age: 12 to 17 years   2011                   82.8   
2      Age: 12 to 17 years   2012                   82.1   
3      Age: 12 to 17 years   2013                   82.0   
4      Age: 12 to 17 years   2014                   83.3   

   Use of public transport  
0                     10.0  
1                      9.8  
2                      8.9  
3                      9.8  
4                      8.9

# Part 1 Data Import

In [47]:
import pandas as pd

In [48]:
# read dataset 1
df1 = pd.read_csv('Project_dataset_1.csv')
df1.columns.values[0] = 'Personal characteristics'
df1

Personal characteristics    Topic  Traffic participation  \
0                                         NaN      Sex  Total male and female   
1                    Personal characteristics  Periods                      %   
2                               Total persons     2018                   82.9   
3                               Total persons     2019                   81.9   
4                               Total persons     2020                   73.8   
5                               Total persons     2021                   78.1   
6                               Total persons     2022                   79.7   
7                          Age: 6 to 11 years     2018                   90.6   
8                          Age: 6 to 11 years     2019                   88.9   
9                          Age: 6 to 11 years     2020                   80.0   
10                         Age: 6 to 11 years     2021                   85.6   
11                         Age: 6 to 11 years     2022                   85.2   
12                        Age: 12 to 17 years     2018                   86.3   
13                        Age: 12 to 17 years     2019                   86.4   
14                        Age: 12 to 17 years     2020                   73.5   
15                        Age: 12 to 17 years     2021                   79.0   
16                        Age: 12 to 17 years     2022                   82.9   
17                        Age: 18 to 24 years     2018                   86.8   
18                        Age: 18 to 24 years     2019                   84.6   
19                        Age: 18 to 24 years     2020                   74.6   
20                        Age: 18 to 24 years     2021                   78.3   
21                        Age: 18 to 24 years     2022                   81.2   
22                        Age: 25 to 34 years     2018                   88.6   
23                        Age: 25 to 34 years     2019                   86.6   
24                        Age: 25 to 34 years     2020                   77.2   
25                        Age: 25 to 34 years     2021                   80.3   
26                        Age: 25 to 34 years     2022                   81.3   
27                        Age: 35 to 49 years     2018                   86.7   
28                        Age: 35 to 49 years     2019                   86.7   
29                        Age: 35 to 49 years     2020                   78.7   
30                        Age: 35 to 49 years     2021                   81.2   
31                        Age: 35 to 49 years     2022                   82.7   
32                        Age: 50 to 64 years     2018                   83.5   
33                        Age: 50 to 64 years     2019                   82.9   
34                        Age: 50 to 64 years     2020                   76.8   
35                        Age: 50 to 64 years     2021                   80.7   
36                        Age: 50 to 64 years     2022                   81.6   
37                        Age: 65 to 74 years     2018                   74.9   
38                        Age: 65 to 74 years     2019                   73.5   
39                        Age: 65 to 74 years     2020                   67.8   
40                        Age: 65 to 74 years     2021                   74.3   
41                        Age: 65 to 74 years     2022                   76.7   
42                     Age: 75 years or older     2018                   59.8   
43                     Age: 75 years or older     2019                   59.4   
44                     Age: 75 years or older     2020                   51.6   
45                     Age: 75 years or older     2021                   59.7   
46                     Age: 75 years or older     2022                   62.6   
47          Equivalised income: 1st 20%-group     2018                   77.1   
48          Equivalised income: 1st 20%-group     20

In [49]:
# read dataset 2
df2 = pd.read_csv('Personenmobiliteit__aandeel_verkeersdeelnemers__persoonskenmerken_2010_2017_gescheiden_door_lijstscheidingsteken.csv', sep=";")
df2

Personenmobiliteit; aandeel verkeersdeelnemers, persoonskenmerken 2010-2017  \
0                                                  NaN                            
1                                                  NaN                            
2                                                  NaN                            
3                                                  NaN                            
4                                    Persoonskenmerken                            
5                                      Totaal personen                            
6                                      Totaal personen                            
7                                      Totaal personen                            
8                                      Totaal personen                            
9                                      Totaal personen                            
10                                     Totaal personen                            
11                                     Totaal personen                            
12                                     Totaal personen                            
13                       Actueel onderwijsniveau: laag                            
14                       Actueel onderwijsniveau: laag                            
15                       Actueel onderwijsniveau: laag                            
16                       Actueel onderwijsniveau: laag                            
17                       Actueel onderwijsniveau: laag                            
18                       Actueel onderwijsniveau: laag                            
19                       Actueel onderwijsniveau: laag                            
20                       Actueel onderwijsniveau: laag                            
21                 Actueel onderwijsniveau: middelbaar                            
22                 Actueel onderwijsniveau: middelbaar                            
23                 Actueel onderwijsniveau: middelbaar                            
24                 Actueel onderwijsniveau: middelbaar                            
25                 Actueel onderwijsniveau: middelbaar                            
26                 Actueel onderwijsniveau: middelbaar                            
27                 Actueel onderwijsniveau: middelbaar                            
28                 Actueel onderwijsniveau: middelbaar                            
29                       Actueel onderwijsniveau: hoog                            
30                       Actueel onderwijsniveau: hoog                            
31                       Actueel onderwijsniveau: hoog                            
32                       Actueel onderwijsniveau: hoog                            
33                       Actueel onderwijsniveau: hoog                            
34                       Actueel onderwijsniveau: hoog                            
35                       Actueel onderwijsniveau: hoog                            
36                       Actueel onderwijsniveau: hoog                            
37                      Migratieachtergrond: Nederland                            
38                      Migratieachtergrond: Nederland                            
39                      Migratieachtergrond: Nederland                            
40                      Migratieachtergrond: Nederland                            
41                      Migratieachtergrond: Nederland                            
42                      Migratieachtergrond: Nederland                            
43                      Migratieachtergrond: Nederland                            
44                      Migratieachtergrond: Nederland                            
45                        Migratieachtergrond: westers                            
46                        Migratieachtergrond: westers                            
47                  

In [50]:
# translate dataset 2 from dutch to english
df2 = df2.replace({'Actueel onderwijsniveau: laag': 'Level of educ: 1. Low'})
df2 = df2.replace({'Actueel onderwijsniveau: middelbaar': 'Level of educ: 2 Intermediate'})
df2 = df2.replace({'Actueel onderwijsniveau: hoog': 'Level of educ: 3 High'})
df2 = df2.replace({'Migratieachtergrond: westers': 'Background: western'})
df2 = df2.replace({'Migratieachtergrond: niet-westers': 'Background: non-western'})
df2 = df2.replace({'Migratieachtergrond: Nederland': 'Background: Dutch'})
df2 = df2.replace({'Leeftijd: 0 tot 12 jaar': 'Age: 6 to 11 years'})
df2 = df2.replace({'Leeftijd: 12 jaar of ouder': 'Age: 12 years or older'})
df2 = df2.replace({'Leeftijd: 12 tot 18 jaar': 'Age: 12 to 17 years'})
df2 = df2.replace({'Leeftijd: 18 tot 25 jaar': 'Age: 18 to 24 years'})
df2 = df2.replace({'Leeftijd: 25 tot 35 jaar': 'Age: 25 to 34 years'})
df2 = df2.replace({'Leeftijd: 35 tot 50 jaar': 'Age: 35 to 49 years'})
df2 = df2.replace({'Leeftijd: 50 tot 65 jaar': 'Age: 50 to 64 years'})
df2 = df2.replace({'Leeftijd: 65 tot 75 jaar': 'Age: 65 to 74 years'})
df2 = df2.replace({'Leeftijd: 75 jaar of ouder': 'Age: 75 years or older'})
df2 = df2.replace({'Categorie: arbeidsongeschikte': 'Particip.: claiming disability benefit'})
df2 = df2.replace({'Categorie: werkzaam 12 tot 30 uur pw': 'Participation: working 12 -29 h. a week'})
df2 = df2.replace({'Categorie: werkzaam >= 30 uur pw': 'Participation: working >30 hours a week'})
df2 = df2.replace({'Categorie: scholier/student': 'Participation:student/scholier'})
df2 = df2.replace({'Participatie:student/scholier': 'Participation:student/scholier'})
df2 = df2.replace({'Categorie: werkloos': 'Participation: unemployed'})
df2 = df2.replace({'Categorie: overig': 'Participation: unemployed, other'})
df2 = df2.replace({'Categorie: gepensioneerd/VUT': 'Participation: retired'})
df2 = df2.replace({'Studenten met OV-weekkaart': 'Students with week pass publ. transp.'})
df2 = df2.replace({'Studenten met OV-weekendkaart': 'Students with weekend pass publ. transp.'})
df2 = df2.replace({'Personen zonder studenten-OV-kaart': 'Persons without student publ.transp.card'})                   
df2 = df2.replace({'Besteedbaar inkomen: 1e 20%-groep': 'Equivalised income: 1th 20%-group'})
df2 = df2.replace({'Besteedbaar inkomen: 2e 20%-groep': 'Equivalised income: 2th 20%-group'})
df2 = df2.replace({'Besteedbaar inkomen: 3e 20%-groep': 'Equivalised income: 3th 20%-group'})
df2 = df2.replace({'Besteedbaar inkomen: 4e 20%-groep': 'Equivalised income: 4th 20%-group'})
df2 = df2.replace({'Besteedbaar inkomen: 5e 20%-groep': 'Equivalised income: 5th 20%-group'})
df2 = df2.replace({'Bezit rijbewijs: geen auto beschikbaar': 'Drivers license, no passenger car in hh'})
df2 = df2.replace({'Bezit rijbewijs: soms auto beschikbaar': 'Drivers license, passenger car in hh'})
df2 = df2.replace({'Bezit rijbewijs: bezit auto': 'Drivers license, passenger car, in own name'})
df2 = df2.replace({'Hoogte inkomen: tot 10 000 euro': 'Income level: up to 10 000 euros'})
df2 = df2.replace({'Hoogte inkomen: 10 000 tot 20 000 euro': 'Income level: 10 000 to 20 000 euros'})
df2 = df2.replace({'Hoogte inkomen: 20 000 tot 30 000 euro': 'Income level: 20 000 to 30 000 euros'})
df2 = df2.replace({'Hoogte inkomen: 30 000 tot 40 000 euro': 'Income level: 30 000 to 40 000 euros'})
df2 = df2.replace({'Hoogte inkomen: 40 000 tot 50 000 euro': 'Income level: 40 000 to 50 000 euros'})
df2 = df2.replace({'Hoogte inkomen: 50 000 euro of meer': 'Income level: 50 000 euros or more'})

df2

Personenmobiliteit; aandeel verkeersdeelnemers, persoonskenmerken 2010-2017  \
0                                                  NaN                            
1                                                  NaN                            
2                                                  NaN                            
3                                                  NaN                            
4                                    Persoonskenmerken                            
5                                      Totaal personen                            
6                                      Totaal personen                            
7                                      Totaal personen                            
8                                      Totaal personen                            
9                                      Totaal personen                            
10                                     Totaal personen                            
11                                     Totaal personen                            
12                                     Totaal personen                            
13                               Level of educ: 1. Low                            
14                               Level of educ: 1. Low                            
15                               Level of educ: 1. Low                            
16                               Level of educ: 1. Low                            
17                               Level of educ: 1. Low                            
18                               Level of educ: 1. Low                            
19                               Level of educ: 1. Low                            
20                               Level of educ: 1. Low                            
21                       Level of educ: 2 Intermediate                            
22                       Level of educ: 2 Intermediate                            
23                       Level of educ: 2 Intermediate                            
24                       Level of educ: 2 Intermediate                            
25                       Level of educ: 2 Intermediate                            
26                       Level of educ: 2 Intermediate                            
27                       Level of educ: 2 Intermediate                            
28                       Level of educ: 2 Intermediate                            
29                               Level of educ: 3 High                            
30                               Level of educ: 3 High                            
31                               Level of educ: 3 High                            
32                               Level of educ: 3 High                            
33                               Level of educ: 3 High                            
34                               Level of educ: 3 High                            
35                               Level of educ: 3 High                            
36                               Level of educ: 3 High                            
37                                   Background: Dutch                            
38                                   Background: Dutch                            
39                                   Background: Dutch                            
40                                   Background: Dutch                            
41                                   Background: Dutch                            
42                                   Background: Dutch                            
43                                   Background: Dutch                            
44                                   Background: Dutch                            
45                                 Background: western                            
46                                 Background: western                            
47                  

In [51]:
df2 = df2.replace({'Persoonskenmerken': 'Personal characteristics'})
df2 = df2.replace({'Perioden': 'Periods'})
df2 = df2.replace({'Geslacht': 'Sex'})
df2 = df2.replace({'Onderwerp': 'Topic'})
df2 = df2.replace({'Verkeersdeelname': 'Traffic participation'})
df2 = df2.replace({'Totaal mannen en vrouwen': 'Total male and female'})
df2 = df2.replace({'Gebruik van het openbaar vervoer': 'Use of public transport'})
df2 = df2.replace({'Mannen': 'Male'})
df2 = df2.replace({'Vrouwen': 'Female'})
df2 = df2.replace({'Totaal personen': 'Total persons'})
df2

Personenmobiliteit; aandeel verkeersdeelnemers, persoonskenmerken 2010-2017  \
0                                                  NaN                            
1                                                  NaN                            
2                                                  NaN                            
3                                                  NaN                            
4                             Personal characteristics                            
5                                        Total persons                            
6                                        Total persons                            
7                                        Total persons                            
8                                        Total persons                            
9                                        Total persons                            
10                                       Total persons                            
11                                       Total persons                            
12                                       Total persons                            
13                               Level of educ: 1. Low                            
14                               Level of educ: 1. Low                            
15                               Level of educ: 1. Low                            
16                               Level of educ: 1. Low                            
17                               Level of educ: 1. Low                            
18                               Level of educ: 1. Low                            
19                               Level of educ: 1. Low                            
20                               Level of educ: 1. Low                            
21                       Level of educ: 2 Intermediate                            
22                       Level of educ: 2 Intermediate                            
23                       Level of educ: 2 Intermediate                            
24                       Level of educ: 2 Intermediate                            
25                       Level of educ: 2 Intermediate                            
26                       Level of educ: 2 Intermediate                            
27                       Level of educ: 2 Intermediate                            
28                       Level of educ: 2 Intermediate                            
29                               Level of educ: 3 High                            
30                               Level of educ: 3 High                            
31                               Level of educ: 3 High                            
32                               Level of educ: 3 High                            
33                               Level of educ: 3 High                            
34                               Level of educ: 3 High                            
35                               Level of educ: 3 High                            
36                               Level of educ: 3 High                            
37                                   Background: Dutch                            
38                                   Background: Dutch                            
39                                   Background: Dutch                            
40                                   Background: Dutch                            
41                                   Background: Dutch                            
42                                   Background: Dutch                            
43                                   Background: Dutch                            
44                                   Background: Dutch                            
45                                 Background: western                            
46                                 Background: western                            
47                  

In [52]:
# remove unnecessary rows
df2.columns = df2.iloc[2]
df2 = df2.drop(index=[0, 1, 2])
df2

2                                            NaN    Topic  \
3                                            NaN      Sex   
4                       Personal characteristics  Periods   
5                                  Total persons     2010   
6                                  Total persons     2011   
7                                  Total persons     2012   
8                                  Total persons     2013   
9                                  Total persons     2014   
10                                 Total persons     2015   
11                                 Total persons     2016   
12                                 Total persons     2017   
13                         Level of educ: 1. Low     2010   
14                         Level of educ: 1. Low     2011   
15                         Level of educ: 1. Low     2012   
16                         Level of educ: 1. Low     2013   
17                         Level of educ: 1. Low     2014   
18                         Level of educ: 1. Low     2015   
19                         Level of educ: 1. Low     2016   
20                         Level of educ: 1. Low     2017   
21                 Level of educ: 2 Intermediate     2010   
22                 Level of educ: 2 Intermediate     2011   
23                 Level of educ: 2 Intermediate     2012   
24                 Level of educ: 2 Intermediate     2013   
25                 Level of educ: 2 Intermediate     2014   
26                 Level of educ: 2 Intermediate     2015   
27                 Level of educ: 2 Intermediate     2016   
28                 Level of educ: 2 Intermediate     2017   
29                         Level of educ: 3 High     2010   
30                         Level of educ: 3 High     2011   
31                         Level of educ: 3 High     2012   
32                         Level of educ: 3 High     2013   
33                         Level of educ: 3 High     2014   
34                         Level of educ: 3 High     2015   
35                         Level of educ: 3 High     2016   
36                         Level of educ: 3 High     2017   
37                             Background: Dutch     2010   
38                             Background: Dutch     2011   
39                             Background: Dutch     2012   
40                             Background: Dutch     2013   
41                             Background: Dutch     2014   
42                             Background: Dutch     2015   
43                             Background: Dutch     2016   
44                             Background: Dutch     2017   
45                           Background: western     2010   
46                           Background: western     2011   
47                           Background: western     2012   
48                           Background: western     2013   
49                           Background: western     2014   
50                           Background: western     2015   
51                           Background: western     2016   
52                           Background: western     2017   
53                       Background: non-western     2010   
54                       Background: non-western     2011   
55                       Background: non-western     2012   
56                       Background: non-western     2013   
57                       Background: non-western     2014   
58                       Background: non-western     2015   
59                       Background: non-western     2016   
60                       Background: non-western     2017   
61              Income level: up to 10 000 euros     2010   
62              Income level: up to 10 000 euros     2011   
63              Income level: up to 10 000 euros     2012   
64              Income level: up to 10 000 euros     2013   
65              Income level: up to 10 000 euros     2014   
66              Income level: up to 10 000 euros     2015   
67              Income level: up to

In [53]:
# rename columns of dataset 2 so they match the column names of dataset 1
df2.columns.values[0] = 'Personal characteristics'
df2.columns.values[3] = 'Traffic participation.1'
df2.columns.values[4] = 'Traffic participation.2'
df2.columns.values[6] = 'Use of public transport.1'
df2.columns.values[7] = 'Use of public transport.2'
df2 = df2[df2['Personal characteristics'] != "Categorie: baby's/kleuters"]
df2

2                       Personal characteristics    Topic  \
3                                            NaN      Sex   
4                       Personal characteristics  Periods   
5                                  Total persons     2010   
6                                  Total persons     2011   
7                                  Total persons     2012   
8                                  Total persons     2013   
9                                  Total persons     2014   
10                                 Total persons     2015   
11                                 Total persons     2016   
12                                 Total persons     2017   
13                         Level of educ: 1. Low     2010   
14                         Level of educ: 1. Low     2011   
15                         Level of educ: 1. Low     2012   
16                         Level of educ: 1. Low     2013   
17                         Level of educ: 1. Low     2014   
18                         Level of educ: 1. Low     2015   
19                         Level of educ: 1. Low     2016   
20                         Level of educ: 1. Low     2017   
21                 Level of educ: 2 Intermediate     2010   
22                 Level of educ: 2 Intermediate     2011   
23                 Level of educ: 2 Intermediate     2012   
24                 Level of educ: 2 Intermediate     2013   
25                 Level of educ: 2 Intermediate     2014   
26                 Level of educ: 2 Intermediate     2015   
27                 Level of educ: 2 Intermediate     2016   
28                 Level of educ: 2 Intermediate     2017   
29                         Level of educ: 3 High     2010   
30                         Level of educ: 3 High     2011   
31                         Level of educ: 3 High     2012   
32                         Level of educ: 3 High     2013   
33                         Level of educ: 3 High     2014   
34                         Level of educ: 3 High     2015   
35                         Level of educ: 3 High     2016   
36                         Level of educ: 3 High     2017   
37                             Background: Dutch     2010   
38                             Background: Dutch     2011   
39                             Background: Dutch     2012   
40                             Background: Dutch     2013   
41                             Background: Dutch     2014   
42                             Background: Dutch     2015   
43                             Background: Dutch     2016   
44                             Background: Dutch     2017   
45                           Background: western     2010   
46                           Background: western     2011   
47                           Background: western     2012   
48                           Background: western     2013   
49                           Background: western     2014   
50                           Background: western     2015   
51                           Background: western     2016   
52                           Background: western     2017   
53                       Background: non-western     2010   
54                       Background: non-western     2011   
55                       Background: non-western     2012   
56                       Background: non-western     2013   
57                       Background: non-western     2014   
58                       Background: non-western     2015   
59                       Background: non-western     2016   
60                       Background: non-western     2017   
61              Income level: up to 10 000 euros     2010   
62              Income level: up to 10 000 euros     2011   
63              Income level: up to 10 000 euros     2012   
64              Income level: up to 10 000 euros     2013   
65              Income level: up to 10 000 euros     2014   
66              Income level: up to 10 000 euros     2015   
67              Income level: up to

In [54]:
# merge dataset 1 and 2 to have the data from 2010 to 2022 combined
df2.reset_index(drop=True, inplace=True)
merged_df = pd.concat([df1, df2], ignore_index=True)
merged_df

Personal characteristics    Topic  \
0                                            NaN      Sex   
1                       Personal characteristics  Periods   
2                                  Total persons     2018   
3                                  Total persons     2019   
4                                  Total persons     2020   
5                                  Total persons     2021   
6                                  Total persons     2022   
7                             Age: 6 to 11 years     2018   
8                             Age: 6 to 11 years     2019   
9                             Age: 6 to 11 years     2020   
10                            Age: 6 to 11 years     2021   
11                            Age: 6 to 11 years     2022   
12                           Age: 12 to 17 years     2018   
13                           Age: 12 to 17 years     2019   
14                           Age: 12 to 17 years     2020   
15                           Age: 12 to 17 years     2021   
16                           Age: 12 to 17 years     2022   
17                           Age: 18 to 24 years     2018   
18                           Age: 18 to 24 years     2019   
19                           Age: 18 to 24 years     2020   
20                           Age: 18 to 24 years     2021   
21                           Age: 18 to 24 years     2022   
22                           Age: 25 to 34 years     2018   
23                           Age: 25 to 34 years     2019   
24                           Age: 25 to 34 years     2020   
25                           Age: 25 to 34 years     2021   
26                           Age: 25 to 34 years     2022   
27                           Age: 35 to 49 years     2018   
28                           Age: 35 to 49 years     2019   
29                           Age: 35 to 49 years     2020   
30                           Age: 35 to 49 years     2021   
31                           Age: 35 to 49 years     2022   
32                           Age: 50 to 64 years     2018   
33                           Age: 50 to 64 years     2019   
34                           Age: 50 to 64 years     2020   
35                           Age: 50 to 64 years     2021   
36                           Age: 50 to 64 years     2022   
37                           Age: 65 to 74 years     2018   
38                           Age: 65 to 74 years     2019   
39                           Age: 65 to 74 years     2020   
40                           Age: 65 to 74 years     2021   
41                           Age: 65 to 74 years     2022   
42                        Age: 75 years or older     2018   
43                        Age: 75 years or older     2019   
44                        Age: 75 years or older     2020   
45                        Age: 75 years or older     2021   
46                        Age: 75 years or older     2022   
47             Equivalised income: 1st 20%-group     2018   
48             Equivalised income: 1st 20%-group     2019   
49             Equivalised income: 1st 20%-group     2020   
50             Equivalised income: 1st 20%-group     2021   
51             Equivalised income: 1st 20%-group     2022   
52             Equivalised income: 2nd 20%-group     2018   
53             Equivalised income: 2nd 20%-group     2019   
54             Equivalised income: 2nd 20%-group     2020   
55             Equivalised income: 2nd 20%-group     2021   
56             Equivalised income: 2nd 20%-group     2022   
57             Equivalised income: 3rd 20%-group     2018   
58             Equivalised income: 3rd 20%-group     2019   
59             Equivalised income: 3rd 20%-group     2020   
60             Equivalised income: 3rd 20%-group     2021   
61             Equivalised income: 3rd 20%-group     2022   
62             Equivalised income: 4th 20%-group     2018   
63             Equivalised income: 4th 20%-group     2019   
64             Equivalised income: 4th 20%-group     2020  

In [55]:
# sort the merged dataset so the same sociodemographic groups of the two data sets are underneath each other
merged_df = merged_df.sort_values(by=['Personal characteristics', 'Topic'])

# drop unnecessary columns
columns_to_drop = ['Traffic participation.1', 'Traffic participation.2', 'Use of public transport.1', 'Use of public transport.2' ]
merged_df = merged_df.drop(columns_to_drop, axis=1)
merged_df = merged_df.drop([0, 1, 137, 138, 139, 460])

In [56]:
#
merged_df['Traffic participation'] = merged_df['Traffic participation'].str.replace(',','.', regex = False)
merged_df['Use of public transport'] = merged_df['Use of public transport'].str.replace(',','.', regex = False)

for row in range(merged_df.shape[0]):
    for column in range(2,merged_df.shape[1]):
        merged_df.iloc[row,column] = pd.to_numeric(merged_df.iloc[row,column], errors='coerce')
        merged_df

In [57]:
# show full merged dataset
pd.set_option('display.max_rows', None)
merged_df

Personal characteristics Topic Traffic participation  \
260                          Age: 12 to 17 years  2010                  82.1   
261                          Age: 12 to 17 years  2011                  82.8   
262                          Age: 12 to 17 years  2012                  82.1   
263                          Age: 12 to 17 years  2013                    82   
264                          Age: 12 to 17 years  2014                  83.3   
265                          Age: 12 to 17 years  2015                  81.3   
266                          Age: 12 to 17 years  2016                  81.4   
267                          Age: 12 to 17 years  2017                  80.5   
12                           Age: 12 to 17 years  2018                  86.3   
13                           Age: 12 to 17 years  2019                  86.4   
14                           Age: 12 to 17 years  2020                  73.5   
15                           Age: 12 to 17 years  2021                  79.0   
16                           Age: 12 to 17 years  2022                  82.9   
244                       Age: 12 years or older  2010                  79.2   
245                       Age: 12 years or older  2011                  79.1   
246                       Age: 12 years or older  2012                  78.8   
247                       Age: 12 years or older  2013                    79   
248                       Age: 12 years or older  2014                  79.4   
249                       Age: 12 years or older  2015                  78.5   
250                       Age: 12 years or older  2016                  78.1   
251                       Age: 12 years or older  2017                  77.7   
268                          Age: 18 to 24 years  2010                  84.9   
269                          Age: 18 to 24 years  2011                  83.9   
270                          Age: 18 to 24 years  2012                    83   
271                          Age: 18 to 24 years  2013                  83.5   
272                          Age: 18 to 24 years  2014                  83.6   
273                          Age: 18 to 24 years  2015                  82.3   
274                          Age: 18 to 24 years  2016                  82.4   
275                          Age: 18 to 24 years  2017                  83.1   
17                           Age: 18 to 24 years  2018                  86.8   
18                           Age: 18 to 24 years  2019                  84.6   
19                           Age: 18 to 24 years  2020                  74.6   
20                           Age: 18 to 24 years  2021                  78.3   
21                           Age: 18 to 24 years  2022                  81.2   
276                          Age: 25 to 34 years  2010                    85   
277                          Age: 25 to 34 years  2011                    86   
278                          Age: 25 to 34 years  2012                  85.3   
279                          Age: 25 to 34 years  2013                  84.4   
280                          Age: 25 to 34 years  2014                  84.6   
281                          Age: 25 to 34 years  2015                  83.9   
282                          Age: 25 to 34 years  2016                  83.8   
283                          Age: 25 to 34 years  2017                  83.5   
22                           Age: 25 to 34 years  2018                  88.6   
23                           Age: 25 to 34 years  2019                  86.6   
24                           Age: 25 to 34 years  2020                  77.2   
25                           Age: 25 to 34 years  2021                  80.3   
26                           Age: 25 to 34 years  2022                  81.3   
284                          Age: 35 to 49 years  2010                  83.1   
285                          Age: 35 to 49 years  2011                  82.9   
286                     

In [58]:
# save merged dataset to csv
merged_df.to_csv('Project_datasets_merged_filtered.csv', index=False)

# Part 2 Data Processing

**1.1**
covid and public transport

**1.2**

In [2]:
def dictionary(df):
    d = {}
    demographic_group = df['Personal characteristics'].unique()
    for i in range(len(demographic_group)):
        d[demographic_group[i]] = df[df['Personal characteristics']==demographic_group[i]]
    return d

def dict_draw(dictionary, group, sex):
    fig = px.line()
    for i in range(len(group)):
        fig.add_scatter(x=dictionary[group[i]]['Topic'], y=dictionary[group[i]][sex], mode='lines', name=group[i])

    fig.update_layout(xaxis_title='year',
                      yaxis_title=sex)
    
    fig.show()

In [30]:
df = pd.read_csv('Project_datasets_merged_filtered.csv', index_col=None)
df['Topic'] = pd.to_datetime(df['Topic'], format='%Y')
d = dictionary(df)

In [7]:
for i in range(len(d)):
    temp = d[list(d.keys())[i]]
    temp['Traffic participation diff'] = temp['Traffic participation'].diff()
    temp['Use of public transport diff'] = temp['Use of public transport'].diff()
    d[list(d.keys())[i]] = temp

/Users/zhangchenhao/anaconda3/envs/TIL6022/lib/python3.7/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  This is separate from the ipykernel package so we can avoid doing imports until
/Users/zhangchenhao/anaconda3/envs/TIL6022/lib/python3.7/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  after removing the cwd from sys.path.


# Part 3 Data Visualisation

# Employment

We use library plotly to draw the graph

In [8]:
import plotly.express as px

We are studying the impact of COVID-19 on traffic by analyzing traffic data grouped by different employment situations.

In [12]:
particip=['Participation: working >30 hours a week',
       'Participation:student/scholier', 'Participation: unemployed',
       'Particip.: claiming disability benefit', 'Participation: retired',
       'Participation: unemployed, other']

First, let's create a line graph for Traffic Participation by different groups over the years, and then a line graph for the differential of Traffic Participation by year.

In [13]:
dict_draw(d, particip, 'Traffic participation')

In [14]:
dict_draw(d, particip, 'Traffic participation diff')

We can observe that, except for the "Participation: unemployed" group, most other groups show a general trend of a slight increase in Traffic Participation from 2017 to 2018. There is a noticeable decline in 2020 when the COVID-19 pandemic hit, and after the pandemic ended in 2021, Traffic Participation gradually recovered to the levels seen before 2018. However, the "Participation: unemployed" group continues to decrease after the pandemic, possibly due to the economic impact of the pandemic, which requires more time for those without stable income to recover.

From the differential data, we can see that the "Participation: student/scholier" group was the most affected by the pandemic, while the "Participation: claiming disability benefit" group recovered the fastest after the pandemic. This could be because some of the policies, such as online schooling, continued during the pandemic, reducing student travel. In contrast, the disabled community primarily has essential needs that require outdoor activities, so they returned to normal levels more rapidly once pandemic-related policies were lifted.

Secondly, let's create a line chart for the Use of Public Transport by different groups over the years, and a line chart for the Year-on-Year Difference in the Use of Public Transport.

In [15]:
dict_draw(d, particip, 'Use of public transport')

In [16]:
dict_draw(d, particip, 'Use of public transport diff')

It can be observed that, except for the groups with missing data, the Use of Public Transport for all other groups significantly decreased in 2020 due to the impact of the pandemic and did not return to pre-pandemic levels by 2022. This could be attributed to the fact that the pandemic had a more pronounced effect on public transportation. During the pandemic, many individuals who used to rely on public transport may have switched to using private cars or bicycles, or even canceled their trips due to policy changes or safety concerns. These changes in travel habits seem to have persisted beyond the pandemic.

From the difference data chart, we also notice that, like the data on transportation participation, students were most affected by the COVID-19 pandemic. This may be because school policies tend to be more stringent compared to the general population. On the other hand, retired individuals experienced relatively smaller impacts from the pandemic, possibly because their initial reliance on public transportation was lower. However, even when overall transportation participation rates recovered, their return to using public transport remained the slowest. This could be attributed to their relatively older age, which may lead to continued caution regarding the pandemic.

Let's get the view of the standard of before, during and after COVID-19. 

In [19]:
def filter_by_year(data_frame, s, e):
    # Convert the 'Date' column to datetime
    data_frame['Topic'] = pd.to_datetime(data_frame['Topic'])
    
    # Create a boolean mask to select rows in the desired date range (s-e)
    mask = (data_frame['Topic'].dt.year >= s) & (data_frame['Topic'].dt.year <= e)
    
    # Apply the mask to filter the DataFrame
    return data_frame[mask]

def get_average(df, data_type):
    average_line_before = pd.concat([filter_by_year(df, 2010, 2019) for df in list(d.values())]).groupby('Personal characteristics')\
        [data_type].mean().reset_index().rename(columns={data_type: 'Before'})
    average_line_during = pd.concat([filter_by_year(df, 2020, 2021) for df in list(d.values())]).groupby('Personal characteristics')\
        [data_type].mean().reset_index().rename(columns={data_type: 'During'})
    average_line_after = pd.concat([filter_by_year(df, 2022, 2022) for df in list(d.values())]).groupby('Personal characteristics')\
        [data_type].mean().reset_index().rename(columns={data_type: 'After'})
    average_line = pd.concat([average_line_before, average_line_during, average_line_after], axis=1)
    average_line = pd.merge(average_line_before, average_line_during, on='Personal characteristics', how='outer')
    average_line = pd.merge(average_line, average_line_after, on='Personal characteristics', how='outer')
    return average_line

We calculate the average number of each group when it is before, during and after COVID-19

In [34]:
aver_upt = get_average(df, 'Use of public transport')
aver_tp = get_average(df, 'Traffic participation')

/Users/zhangchenhao/anaconda3/envs/TIL6022/lib/python3.7/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



In [41]:
mask = aver_upt['Personal characteristics'].isin(particip)
aver_upt = aver_upt[mask]
aver_tp = aver_tp[mask]

And we draw the bar graph which can show the average standard of each period

In [46]:
import plotly.graph_objs as go
from plotly.subplots import make_subplots

def draw_bar(df):
    fig = make_subplots(rows=1, cols=1)

    # Add bar traces, one for each data column
    fig.add_trace(go.Bar(x=df['Personal characteristics'], y=df['Before'], name='Before'), row=1, col=1)
    fig.add_trace(go.Bar(x=df['Personal characteristics'], y=df['During'], name='During'), row=1, col=1)
    fig.add_trace(go.Bar(x=df['Personal characteristics'], y=df['After'], name='After'), row=1, col=1)

    # Change the bar mode to group
    fig.update_layout(barmode='group')

    # Show the figure
    fig.show()


In [47]:
draw_bar(aver_upt)

In [48]:
draw_bar(aver_tp)

To make it more clear we draw the calculate the difference of each period and show it with the graph.

In [50]:
import copy
def get_diff(data):
    df = copy.deepcopy(data)
    df['drop'] = df['During'] - df['Before']
    df['increace'] = df['After'] - df['During']
    return df

In [61]:
aver_tp = get_diff(aver_tp)
aver_upt = get_diff(aver_upt)

In [58]:
fig = px.bar(aver_tp, x='Personal characteristics', y=['drop', 'increace'], title='Traffic participation')
# Show the plot
fig.show()

In [63]:
fig = px.bar(aver_upt, x='Personal characteristics', y=['drop', 'increace'], title='Use of public transport')
# Show the plot
fig.show()

For the "Traffic participation" graph, we notice that the 'Participation: claiming disability benefit' and 'Participation: student/scholar' groups have shown a significant increase in traffic participation post-COVID. And for 'Participation: retired', 'Participation: unemployed', 'Participation: unemployed, other', and 'Participation: working >30 hours a week', there is a notable drop in traffic participation during or after COVID-19. 

For the "Use of public transport" graph, similarly, the red bars denote an increase and the blue bars a decrease in the use of public transport after COVID-19. There is an increase in public transport usage among 'Participation: student/scholar' and a slight increase in 'Participation: working >30 hours a week'. All other groups ('Participation: claiming disability benefit', 'Participation: retired', 'Participation: unemployed', and 'Participation: unemployed, other') demonstrate a decrease in the use of public transport.

We believe that students and those working more than 30 hours a week may have returned to their regular routines, which involves more travel and use of public transport. Retired individuals and those claiming disability benefits might be more cautious and continue to limit their exposure to public places, including traffic and public transport, even after the pandemic has subsided. The unemployed group's reduced participation in traffic and public transport could be due to fewer job opportunities or a shift to remote working conditions where travel is less necessary.

# Income

In [26]:
# please remove this as soon as the data processing is in de document

In [27]:
pip install cbsodata

Note: you may need to restart the kernel to use updated packages.


In [28]:
import cbsodata
import pandas as pd
import seaborn as sns
import plotly.express as px
import matplotlib.pyplot as plt
import plotly.io as pio
df2 = pd.DataFrame(cbsodata.get_data('83496NED'));

In [29]:
df2 = df2.replace({'Actueel onderwijsniveau: laag': 'Level of educ: 1. Low'})
df2 = df2.replace({'Actueel onderwijsniveau: middelbaar': 'Level of educ: 2 Intermediate'})
df2 = df2.replace({'Actueel onderwijsniveau: hoog': 'Level of educ: 3 High'})
df2 = df2.replace({'Migratieachtergrond: westers': 'Background: western'})
df2 = df2.replace({'Migratieachtergrond: niet-westers': 'Background: non-western'})
df2 = df2.replace({'Migratieachtergrond: Nederland': 'Background: Dutch'})
df2 = df2.replace({'Leeftijd: 0 tot 12 jaar': 'Age: 6 to 11 years'})
df2 = df2.replace({'Leeftijd: 12 jaar of ouder': 'Age: 12 years or older'})
df2 = df2.replace({'Leeftijd: 12 tot 18 jaar': 'Age: 12 to 17 years'})
df2 = df2.replace({'Leeftijd: 18 tot 25 jaar': 'Age: 18 to 24 years'})
df2 = df2.replace({'Leeftijd: 25 tot 35 jaar': 'Age: 25 to 34 years'})
df2 = df2.replace({'Leeftijd: 35 tot 50 jaar': 'Age: 35 to 49 years'})
df2 = df2.replace({'Leeftijd: 50 tot 65 jaar': 'Age: 50 to 64 years'})
df2 = df2.replace({'Leeftijd: 65 tot 75 jaar': 'Age: 65 to 74 years'})
df2 = df2.replace({'Leeftijd: 75 jaar of ouder': 'Age: 75 years or older'})
df2 = df2.replace({'Categorie: arbeidsongeschikte': 'Particip.: claiming disability benefit'})
df2 = df2.replace({'Categorie: werkzaam 12 tot 30 uur pw': 'Participation: working 12 -29 h. a week'})
df2 = df2.replace({'Categorie: werkzaam >= 30 uur pw': 'Participation: working >30 hours a week'})
df2 = df2.replace({'Categorie: scholier/student': 'Participation:student/scholier'})
df2 = df2.replace({'Participatie:student/scholier': 'Participation:student/scholier'})
df2 = df2.replace({'Categorie: werkloos': 'Participation: unemployed'})
df2 = df2.replace({'Categorie: overig': 'Participation: unemployed, other'})
df2 = df2.replace({'Categorie: gepensioneerd/VUT': 'Participation: retired'})
df2 = df2.replace({'Studenten met OV-weekkaart': 'Students with week pass publ. transp.'})
df2 = df2.replace({'Studenten met OV-weekendkaart': 'Students with weekend pass publ. transp.'})
df2 = df2.replace({'Personen zonder studenten-OV-kaart': 'Persons without student publ.transp.card'})                   
df2 = df2.replace({'Besteedbaar inkomen: 1e 20%-groep': 'Equivalised income: 1th 20%-group'})
df2 = df2.replace({'Besteedbaar inkomen: 2e 20%-groep': 'Equivalised income: 2th 20%-group'})
df2 = df2.replace({'Besteedbaar inkomen: 3e 20%-groep': 'Equivalised income: 3th 20%-group'})
df2 = df2.replace({'Besteedbaar inkomen: 4e 20%-groep': 'Equivalised income: 4th 20%-group'})
df2 = df2.replace({'Besteedbaar inkomen: 5e 20%-groep': 'Equivalised income: 5th 20%-group'})
df2 = df2.replace({'Bezit rijbewijs: geen auto beschikbaar': 'Drivers license, no passenger car in hh'})
df2 = df2.replace({'Bezit rijbewijs: soms auto beschikbaar': 'Drivers license, passenger car in hh'})
df2 = df2.replace({'Bezit rijbewijs: bezit auto': 'Drivers license, passenger car, in own name'})
df2 = df2.replace({'Hoogte inkomen: tot 10 000 euro': 'Income level: up to 10 000 euros'})
df2 = df2.replace({'Hoogte inkomen: 10 000 tot 20 000 euro': 'Income level: 10 000 to 20 000 euros'})
df2 = df2.replace({'Hoogte inkomen: 20 000 tot 30 000 euro': 'Income level: 20 000 to 30 000 euros'})
df2 = df2.replace({'Hoogte inkomen: 30 000 tot 40 000 euro': 'Income level: 30 000 to 40 000 euros'})
df2 = df2.replace({'Hoogte inkomen: 40 000 tot 50 000 euro': 'Income level: 40 000 to 50 000 euros'})
df2 = df2.replace({'Hoogte inkomen: 50 000 euro of meer': 'Income level: 50 000 euros or more'})
df2 = df2.rename(columns={"Geslacht": "Sex", "Persoonskenmerken": "PersonalCharacteristics", "Geslacht": "Sex", "Perioden": "Periods", 
                    "Verkeersdeelname_1": "TrafficParticipation_1", "GebruikVanHetOpenbaarVervoer_2": "UseOfPublicTransport_2"});

In [30]:
df1 = pd.DataFrame(cbsodata.get_data('84707ENG'))
df_value = df1[( df1['Population'] == 'Population 6 years or older') 
              & (df1['Margins'] == 'Value') & (df1['Sex'] != 'Sex unknown')];
df_value = df_value.drop(columns=['Margins', 'Population']);

df = pd.concat([df_value, df2], ignore_index=True);

df_sorted = df.sort_values(by=['PersonalCharacteristics', 'Periods'])
df_sorted = df_sorted[(df_sorted['Sex'] == 'Total male and female')]
df_sorted

ID                    Sex                   PersonalCharacteristics  \
10    30  Total male and female                       Age: 12 to 17 years   
11    31  Total male and female                       Age: 12 to 17 years   
12    32  Total male and female                       Age: 12 to 17 years   
13    33  Total male and female                       Age: 12 to 17 years   
14    34  Total male and female                       Age: 12 to 17 years   
15    45  Total male and female                       Age: 18 to 24 years   
16    46  Total male and female                       Age: 18 to 24 years   
17    47  Total male and female                       Age: 18 to 24 years   
18    48  Total male and female                       Age: 18 to 24 years   
19    49  Total male and female                       Age: 18 to 24 years   
20    60  Total male and female                       Age: 25 to 34 years   
21    61  Total male and female                       Age: 25 to 34 years   
22    62  Total male and female                       Age: 25 to 34 years   
23    63  Total male and female                       Age: 25 to 34 years   
24    64  Total male and female                       Age: 25 to 34 years   
25    75  Total male and female                       Age: 35 to 49 years   
26    76  Total male and female                       Age: 35 to 49 years   
27    77  Total male and female                       Age: 35 to 49 years   
28    78  Total male and female                       Age: 35 to 49 years   
29    79  Total male and female                       Age: 35 to 49 years   
30    90  Total male and female                       Age: 50 to 64 years   
31    91  Total male and female                       Age: 50 to 64 years   
32    92  Total male and female                       Age: 50 to 64 years   
33    93  Total male and female                       Age: 50 to 64 years   
34    94  Total male and female                       Age: 50 to 64 years   
5     15  Total male and female                        Age: 6 to 11 years   
6     16  Total male and female                        Age: 6 to 11 years   
7     17  Total male and female                        Age: 6 to 11 years   
8     18  Total male and female                        Age: 6 to 11 years   
9     19  Total male and female                        Age: 6 to 11 years   
35   105  Total male and female                       Age: 65 to 74 years   
36   106  Total male and female                       Age: 65 to 74 years   
37   107  Total male and female                       Age: 65 to 74 years   
38   108  Total male and female                       Age: 65 to 74 years   
39   109  Total male and female                       Age: 65 to 74 years   
40   120  Total male and female                    Age: 75 years or older   
41   121  Total male and female                    Age: 75 years or older   
42   122  Total male and female                    Age: 75 years or older   
43   123  Total male and female                    Age: 75 years or older   
44   124  Total male and female                    Age: 75 years or older   
45   135  Total male and female                         Background: Dutch   
46   136  Total male and female                         Background: Dutch   
47   137  Total male and female                         Background: Dutch   
48   138  Total male and female                         Background: Dutch   
49   139  Total male and female                         Background: Dutch   
55   165  Total male and female                   Background: non-western   
56   166  Total male and female                   Background: non-western   
57   167  Total male and female                   Background: non-western   
58   168  Total male and female                   Background: non-western   
59   169  Total male and female                   Background: non-western   
50   150  Total male and female                       Background: western   
51

In [31]:
mask = df_sorted['PersonalCharacteristics'].str.startswith('Equi');
df_income = df_sorted[mask];
fig = px.line(df_income, y='UseOfPublicTransport_2', x='Periods', color='PersonalCharacteristics')
fig.show()

In [32]:
mask = df_sorted['PersonalCharacteristics'].str.startswith(('Driv','No'));
df_drivers_licence = df_sorted[mask];
fig = px.line(df_drivers_licence, y='UseOfPublicTransport_2', x='Periods', color='PersonalCharacteristics')
fig.show()

**Literature** 

Paulley et al. (2006) identify four key relationships between income and public transport usage:

1. An increase in income will, depending upon the level of income, lead to an increase in car ownership and so car availability, or to an increase in public transport use.
2. An increase in car ownership/availability will, other things being equal, lead to a reduction in the demand for public transport modes.
3. The sign and magnitude of demand elasticities for public transport with respect to car availability and income will vary depending upon the income levels.
4. Income growth can be expected to increase average trip length.

Holmgren (2013) found that income's direct influence on public transport demand and income's indirect influence through car ownership work in opposite directions, so that the total income effect is close to zero. 

In the USA, poorer people live closer to city centers because of their dependency on cheap public transportation, while richer people can afford to travel by car to their work (Glaeser et al., 2008). Van Goeverden et al. (2006) state that also in the Netherlands low-income people may not be able to afford alternative transport and thus are dependent on public transportation.

MuConsult (1992) shows that train use in the Netherlands does increase with income, controlling for other variables such as car ownership, residential location, and educational level. Buehler and Pucher (2012) express that the lowest income percentile group travels the most with public transport, while the second, third, and fourth income groups have almost identical travel percentages.

During COVID, working from home was a privilege of higher-income jobs (Tirachini & Cats, 2020). Tirachini and Cats (2020) continue by stating that the amount of trips on public transportation fell by between 30% and 40% for people in the lowest income households, while the decrease in public transportation use was greater than 70% for the highest income households in their survey. Almlöf et al. (2021) also found that the higher the income, the higher the drop in public transport usage during the pandemic.

**Analysis** 

The investigated data supports the findings of the literature. The poor travel more by public transport. They started using PT faster after COVID than members of higher income groups, propably because they did not have the possiblity to work at home. 
Car ownership and driver licence ownerhip are negatively correlated with PT usage. People with a car in their household switched to the car during the pandemic and did not all return to PT after COVID. In the data we see that people without a driver licence returned faster to PT than people with a driver licence but without a car. Possibly that people with a driver licence have bought a car of use rent-a-car services more often.


**References**
- Almlöf, E., Rubensson, I., Cebecauer, M., & Jenelius, E. (2021). [Who continued traveling by public transport during COVID-19? Socioeconomic factors explaining travel behavior in Stockholm 2020 based on smart card data.](https://doi.org/10.1186/s12544-021-00488-0) European Transport Research Review, 13(1).
- Buehler, R., & Pucher, J. (2012). [Demand for public transport in Germany and the USA: An analysis of rider characteristics.](https://doi.org/10.1080/01441647.2012.707695) Transport Reviews, 32(5), 541–567.
- Glaeser, E. L., Kahn, M. E., & Rappaport, J. (2008). [Why do the poor live in cities? The role of public transportation.](https://doi.org/10.1016/j.jue.2006.12.004) Journal of Urban Economics, 63(1), 1–24.
- Holmgren, J. (2013). [An analysis of the determinants of local public transport demand focusing the effects of income changes.](https://doi.org/10.1007/s12544-013-0094-0) European Transport Research Review, 5(2), 101–107.
- Koehl, A. (2020). [Urban transport and COVID-19: challenges and prospects in low- and middle-income countries.](https://doi.org/10.1080/23748834.2020.1791410) Cities & Health, 5(sup1), S185–S190.
- MuConsult. (1992). [Inkomen en treingebruik naar motief [Income and Train Use by Trip Purpose]](https://www.example.com). Amersfoort: MuConsult.
- Paulley, N., Balcombe, R. J., Mackett, R., Titheridge, H., Preston, J., Wardman, M., Shires, J., & White, P. (2006). [The demand for public transport: The effects of fares, quality of service, income and car ownership.](https://doi.org/10.1016/j.tranpol.2005.12.004) Transport Policy, 13(4), 295–306.
- Shadmi, E., Chen, Y., Dourado, I., Faran-Perach, I., Furler, J., Halmai, P., Hanvoravongchai, P., Obando, C., Petrosyan, V., Rao, K. D., Ruano, A. L., Shi, L., De Souza, L. E., Spitzer-Shohat, S., Sturgiss, E., Suphanchaimat, R., Uribe, M. V., & Willems, S. (2020). [Health equity and COVID-19: global perspectives.](https://doi.org/10.1186/s12939-020-01218-z) International Journal for Equity in Health, 19(1).
- Tirachini, A., & Cats, O. (2020). [COVID-19 and public transportation: current assessment, prospects, and research needs.](https://doi.org/10.5038/2375-0901.22.1.1) Journal of Public Transportation, 22(1).
- Van Goeverden, C., Rietveld, P., Koelemeijer, J., & Peeters, P. (2006). [Subsidies in public transport.](https://www.openstarts.units.it/bitstream/10077/5892/1/vanGoeverden_et_al_ET32.pdf) European Transport, 32(32), 5–25.


# Age

In [33]:
df['gap with mean']=0
for i in range(0, df.shape[0]):
    ref_year = df.loc[i, 'Topic']
    ref_value = df[(df['Personal characteristics'] == 'Total persons') & (df['Topic'] == ref_year)].iloc[0, 3]
    df.loc[i, 'gap with mean'] = df.loc[i, 'Use of public transport'] - ref_value

df

KeyError: 'Topic'

In [ ]:
df[(df['Personal characteristics'] == 'Age: 12 years or older') & (df['Topic'] == '2018')].shape

In [ ]:
df['comparison to 2018 level']=0
for i in range(df.shape[0]):
    ref_group = df.loc[i, 'Personal characteristics']
    if df[(df['Personal characteristics'] == ref_group) & (df['Topic'] == 2018)].shape[0] == 0:
        df.loc[i, 'comparison to 2018 level'] = 'NaN'
    else:
        ref_value = df[(df['Personal characteristics'] == ref_group) & (df['Topic'] == 2018)].iloc[0, 3]
        comparison_value = df.loc[i, 'Use of public transport']
        df.loc[i, 'comparison to 2018 level'] = comparison_value - ref_value

df

In [ ]:
df['comparison to 2018 level (%)']=0
for i in range( df.shape[0]):
    ref_group = df.loc[i, 'Personal characteristics']
    if df[(df['Personal characteristics'] == ref_group) & (df['Topic'] == 2018)].shape[0] == 0:
        df.loc[i, 'comparison to 2018 level'] = 'NaN'
    else:
        ref_value = df[(df['Personal characteristics'] == ref_group) & (df['Topic'] == 2018)].iloc[0, 3]
        comparison_value = df.loc[i, 'Use of public transport']
        df.loc[i, 'comparison to 2018 level (%)'] = ((comparison_value - ref_value)/ref_value)*100
df['comparison to 2018 level (%)']

In [ ]:
# Creation of a dataset only encompassing the Total 
mask = df['Personal characteristics'].str.startswith('Total')
df_total = df[mask]

In [ ]:
# Creation of a dataset focusing on age groups, with one dataset also having the total group for comparison
mask = df['Personal characteristics'].str.startswith('Age')
df_age = df[mask]
df_age2 = pd.concat([df_total, df_age])

In [ ]:
px.line(data_frame=df_age2, x="Topic", y="Use of public transport", color='Personal characteristics')

#### Observations :
The drop of traffic during covid is more sharp for younger categories (age < 35 years) compared to the other age groups. Maybe this can be explained by the closing of schools and universities. Maybe younger workers also tend to use remote working more than older ones.
Trend before covid : stagnation and for some groups slight increase.
The traffic levels have yet to recover to the pre-covid levels.

In [ ]:
px.line(data_frame=df_age, x="Topic", y="gap with mean", color='Personal characteristics')

#### Observations :
What is interesting to note is that for people 35 years old and older, the gap with the mean is quite constant during the pandemic, and is even reduced for some categories. Post-covid gaps are generally less important but still negative. Thus, we can imagine that people aged 35 and more were less impacted by the pandemic than other categories. This can maybe be explained by the fact that a lot of younger persons, who make up the bulk of public transport traffic, stopped using public transport because of schools and universities closing, whereas certain necessary workplaces were still open.
 

In [ ]:
px.bar(data_frame = df_age2[(df_age2['Topic'] == 2022)], x='Personal characteristics', y='comparison to 2018 level', color='Personal characteristics')

#### Observations :
In terms of raw participation, the group with the sharpest drop is the 18 to 24 years age group. As mentionned before, this can maybe be explained by the fact during the pandemic, many universities were closed, so universities implemented more online courses, which may have reduced the transportation needs of students after the pandemic. Also, maybe people preferred using individual transport means (bikes, cars) during the pandemic to avoid restrictions and kept this habit afterwards.
The second most impacted group are people from 25 to 34 years old. Maybe this can be explained by an increase in remote work after the pandemic
However, we have to keep in mind that the variation is bigger in these groups because they are already the groups with the highest participation

In [ ]:
px.bar(data_frame = df_age2[(df_age2['Topic'] == 2022)], x='Personal characteristics', y='comparison to 2018 level (%)', color='Personal characteristics')

In terms or percentages, the average variation (variation for the total) is - 29%.
Even if the variation was higher in terms of raw numbers for the younger age groups, proportionnaly the drop was bigger for age groups over 35 years old.
One explanation could be that, since these groups were already using public transports less than other groups, people from these older age groups were more likely to stop taking public transport, or at least decrease significantly their use of public transport. Furthermore, maybe these groups have more alternatives to public transport than younger age groups, for example access to a car.

# Students with week or weekend pass

In this part the persons without student public transport card, the students with week pass public transport and the students with weekend pass public transport will be visualized and analyzed. The before COVID-19 situation will be compared with the during COVID-19 situation and a hypothesis on the after COVID-19 situation will be looked into. 

In [44]:
# List of student transportation options
students=['Persons without student publ.transp.card',
       'Students with week pass publ. transp.',
       'Students with weekend pass publ. transp.'] 

First, a line graph is created for Traffic Participation by persons without student public transport card, students with week pass public transport and students with weekend pass public transport over the years. 

In [45]:
dict_draw(d, students, 'Traffic participation')

In the graph is visuable that students with weekend pass public transport have the highest traffic participation. Before, during and COVID-19 this is consistently the case. In 2014 and in 2018 there is a peak in the traffic participation of this sociodemographic group. Persons without student public transport pass tend to participate the least in traffic. In 2018 all these groups have the highest traffic participation in 2018 within the years examined. In 2019 the COVID-19 period started, studying from home became the norm. This can possibly explain the large decrease of traffic participation of all the groups. The data of 2020 stands out. Students with week pass public transport and person without a student public transport card have almost the same participation in this year. While students with weekend pass public transport travel less than before COVID-19, but clearly more than the other groups. In 2021 and 2022 the traffic participation increases for all groups.

The next line graph is for the differential of Traffic Participation of the same groups by year. Showing the differential of Traffic Participation will give a good insight in the differences in public transport participation for the different groups per year. 

In [46]:
dict_draw(d, students, 'Traffic participation diff')

The differential data of 2020 shows that the students with weekend pass public transport experienced the greatest impact. Remarkable is that the differential data shows that the percentage of the participation of students with weekend pass public transport is increasing in 2021 and 2022, while the other two groups had a increase in traffic participation from 2020 to 2021, but from 2021 to 2022 a decrease is visuable. 

In the next line graph the Use of Public Transport by different groups over the years is shown. Unfortunately the data of 2021 and 2022 of students with weekend pass public transport is missing.

In [42]:
dict_draw(d, students, 'Use of public transport')

It can be observed that the Use of Public Transport for students with weekend pass public transport and students with week pass public transport significantly decreased in 2020 due to the impact of the pandemic. In 2022 the Use of Public Transport didn't return to pre-pandemic levels. Also for these demographic groups it is possible that the pandemic had a more pronounced effect on public transportation. Notable is that persons without a student public transport card started to use public transport more from 2018 and it looks like this group almost reach the pre-pandemic levels in 2022.

Below a line graph for the Year-on-Year Difference in the Use of Public Transport for the persons without student public transport card, the students with week pass public transport and the students with weekend pass public transport is shown.

In [43]:
dict_draw(d, students, 'Use of public transport diff')

The difference data chart shows that students (with week pass public transport and probably with weekend pass public transport) were most affected by the COVID-19 pandemic. As mentioned before, students often rely on public transport for studying. An assumption is that Students with week pass public transport tend to not live in the same area as the education location. Students with weekend pass public transport usually live close(r) to the education location. This group uses the public transport card for example for visiting their parents. During COVID-19 studying from home was common. Students with week pass public transport need their public transport pass less than students with weekend pass who still want to visit their parents.
 